libmariadb/CMakeLists.txt

```diff
@@ -4,6 +4,9 @@ INCLUDE_DIRECTORIES(${CC_SOURCE_DIR}/include
 ADD_DEFINITIONS(-D HAVE_COMPRESS)
 ADD_DEFINITIONS(-D LIBMARIADB)
 ADD_DEFINITIONS(-D THREAD)
+IF(APPLE)
+  ADD_DEFINITIONS(-D _XOPEN_SOURCE)
```

Comment:
shouldn't it be `_XOPEN_SOURCE=600` ?

Reply:
I have no idea why it was `600` as I checked `ucontext.h`: 
```
#ifdef _XOPEN_SOURCE
...
#else /* !_XOPEN_SOURCE */
#error The deprecated ucontext routines require _XOPEN_SOURCE to be defined
#endif /* _XOPEN_SOURCE */
```

Also it seems it was producing compilation errors if I set `_XOPEN_SOURCE` to `600`. Setting `-D _XOPEN_SOURCE` just works fine.

Reply:
ok, fair enough